src/lib/registration/sw_mgmt.rb

```diff
@@ -259,30 +260,29 @@ def self.evaluate_product(p, selected, installed)
     # Any product selected to install?
     # @return [Boolean] true if at least one product is selected to install
     def self.product_selected?
-      Pkg.ResolvableProperties("", :product, "").any? { |p| p["status"] == :selected }
+      Y2Packager::Resolvable.any?(kind: :product, status: :selected)
     end
 
     # Any product installed? (e.g. during upgrade)
     # @return [Boolean] true if at least one product is installed
     def self.product_installed?
-      Pkg.ResolvableProperties("", :product, "").any? do |p|
-        p["status"] == :installed || p["status"] == :removed
-      end
+      Y2Packager::Resolvable.any?(kind: :product, status: :installed) ||
+        Y2Packager::Resolvable.any?(kind: :product, status: :removed)
     end
 
     def self.installed_products
       # just for testing/debugging
       return [FAKE_BASE_PRODUCT] if ENV["FAKE_BASE_PRODUCT"]
```

Comment:
The `FAKE_BASE_PRODUCT` is a Hash, this needs to be converted to `Y2Packager::Resolvable` so this always returns the same class.